nomos-services/cryptarchia-consensus/src/lib.rs

```diff
@@ -872,6 +854,206 @@ where
+            TxS::Settings,
+            BS::Settings,
+            NetAdapter::Settings,
+            BlendAdapter::Settings,
+        >,
+        genesis_id: HeaderId,
+        genesis_state: LedgerState,
+        ledger_config: nomos_ledger::Config,
+        leader_config: LeaderConfig,
+        relays: &CryptarchiaConsensusRelays<
+            BlendAdapter,
+            BS,
+            ClPool,
+            ClPoolAdapter,
+            DaPool,
+            DaPoolAdapter,
+            NetAdapter,
+            SamplingBackend,
+            SamplingRng,
+            Storage,
+            TxS,
+        >,
+        block_subscription_sender: &mut broadcast::Sender<Block<ClPool::Item, DaPool::Item>>,
+    ) -> (Cryptarchia, Leader) {
+        match initial_state.recovery_strategy() {
+            CryptarchiaInitialisationStrategy::Genesis => {
+                info!("Building Cryptarchia from genesis...");
+                let leader = Leader::from_genesis(genesis_id, leader_config, ledger_config.clone());
+                let cryptarchia = Cryptarchia::new(genesis_id, genesis_state, ledger_config);
+
+                (cryptarchia, leader)
+            }
+            CryptarchiaInitialisationStrategy::RecoveryFromGenesis(GenesisRecoveryStrategy {
+                tip,
+            }) => {
+                info!("Recovering Cryptarchia with Genesis strategy.");
+                let mut leader =
+                    Leader::from_genesis(genesis_id, leader_config, ledger_config.clone());
+                let cryptarchia = Self::recover_cryptarchia(
+                    genesis_id,
+                    genesis_state,
+                    tip,
+                    &mut leader,
+                    ledger_config,
+                    relays,
+                    block_subscription_sender,
+                )
+                .await;
+
+                (cryptarchia, leader)
+            }
+            CryptarchiaInitialisationStrategy::RecoveryFromSecurity(strategy) => {
+                let SecurityRecoveryStrategy {
+                    tip,
+                    security_block_id,
+                    security_ledger_state,
+                    security_leader_notes,
+                } = *strategy;
+
+                info!("Recovering Cryptarchia with Security strategy.");
+                let mut leader = Leader::new(
+                    security_block_id,
+                    security_leader_notes,
+                    leader_config.nf_sk,
+                    ledger_config.clone(),
+                );
+
+                let cryptarchia = Self::recover_cryptarchia(
+                    security_block_id,
+                    security_ledger_state,
+                    tip,
+                    &mut leader,
+                    ledger_config,
+                    relays,
+                    block_subscription_sender,
+                )
+                .await;
+
+                (cryptarchia, leader)
+            }
+        }
+    }
```

Comment:
Nitpick: This could be refactore as an implementation of `CryptarchiaInitialisationStrategy` (?), with independent methods for each case.

```rust
impl CryptarchiaInitializationStrategy {
  pub fn recover(&self, *args) -> (Cryptarchia, Leader) {
      match self {
          CryptarchiaInitialisationStrategy::Genesis => recover_from_genesis(self, *args),
          ...
      }
  }
}
```

Or at least separated methods for each branch.

Reply:
The thought did cross my mind but I forgot. Thanks!

Reply:
As good as it sounded moving that code to the strategy structs, it propagates generics to them as well which seemed a bit messy.

For now I ended up just separating them into methods for each branch, and they can be easily moved into strategy if we want.

Do you guys consider worth it worth? That is, moving them to the strategy section.